source/lac/trilinos_sparse_matrix.cc

```diff
@@ -126,9 +126,6 @@ namespace TrilinosWrappers
           return;
         }
 
-      // otherwise first flush Trilinos caches
-      matrix->compress ();
-
```

Comment:
That actually took a while to fix, but there are now `mpi/trilinos_sparse_matrix_0[45]` for this.